### DISCUSSION TITLE
[gatsby-plugin-mdx]: Combining/concatenation/transclusion of MDX files

### DISCUSSION THREAD

**Discussion by nikhilxb** [gatsbyjs/gatsby#28398] [Help]
## Summary

I'm trying to break up a large Markdown (.mdx) file into multiple smaller ones, where each one has footnotes:

In `parent.mdx`:
```mdx
import Child1 from "./child1.mdx";
import Child2 from "./child2.mdx";
<Child1 />
<Child2 />
```

In `child1.mdx`:
```mdx
This sentence has a footnote [^foo].
[^foo]: Footnote 1.
```

In `child2.mdx`:
```mdx
This sentence has a footnote [^bar].
[^bar]: Footnote 2.
```

Since each child MDX is compiled then included in parent MDX, this results in *2 separate footnotes sections* rather that 1 unified footnote section, as desired. What I want is true transclusion of the raw text (something like `#include`), ideally a solution that supports recursion (i.e. child can also `#include` a snippet).

### What I've tried

- Using https://github.com/hashicorp/remark-plugins/blob/master/plugins/include-markdown/index.js, but it uses `remark` to parse regular MD files and doesn't seem to work for MDX.
- Writing a custom remark plugin, but I'm not sure how to recursively generate the AST using the functions provided by `gatsby-plugin-mdx`.
- Manually overwriting .mdx `File`'s `node.internal.content` in `onCreateNode` to concatenate the raw text, but `gatsby-plugin-mdx` doesn't seem to be affected.
- Writing a preprocessing plugin as suggested in #15868, but that doesn't work for `gatsby-plugin-mdx` as noted in #23409

Any ideas?

### Motivation

Particularly for large projects, it's useful to break posts into more manageable chunks.

**Comment by FrancoisDF** [comment]
I'm also interested by this feature. I have also tried different solution without success. I know that we can import markdown as a component, but my main concern is that those components will then be part of the bundle. If preprocessing was possible, we could import the markdown content inside a markdown without extra layer of javascript bundling. This will dramatically reduce the bundle size and improve prefromances
